fiftyone/core/session/session.py

```
@@ -1298,6 +1369,70 @@ def _on_refresh(session: Session, state: t.Optional[StateDescription]):
+            sample_id = item.get("sample_id")
+            sel_type = item.get("type", "default")
+            if sel_type not in valid_types:
+                raise ValueError(
+                    f"Invalid selection type '{sel_type}' for sample "
+                    f"'{sample_id}'. Must be one of {valid_types}"
+                )
+            result.append({"sample_id": sample_id, "type": sel_type})
+        else:
+            raise ValueError(
+                f"Invalid sample entry: {item}. Must be a string or dict"
+            )
```

Ruff (0.15.5)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)